app/charts/shared/legend-color.tsx

```diff
@@ -184,22 +188,37 @@ export const LegendColor = memo(function LegendColor(props: LegendColorProps) {
     getLegendItemDimension,
     interactive,
     showTitle,
+    dimensionsById,
   } = props;
   const { colors, getColorLabel } = useChartState() as ColorsChartState;
   const values = colors.domain();
   const groups = useLegendGroups({ chartConfig, segmentDimension, values });
 
   return (
-    <LegendColorContent
-      groups={groups}
-      getColor={(v) => colors(v)}
-      getLabel={getColorLabel}
-      getItemDimension={getLegendItemDimension}
-      symbol={symbol}
-      showTitle={showTitle}
-      interactive={interactive}
-      numberOfOptions={values.length}
-    />
+    <Box>
+      {showTitle &&
+        isSegmentInConfig(chartConfig) &&
+        chartConfig.fields.segment &&
+        dimensionsById && (
+          <Typography
+            data-testId="colorLegendTitle"
+            component="div"
+            variant="subtitle1"
+            color={theme.palette.primary.main}
+          >
+            {dimensionsById[chartConfig.fields.segment.componentId]?.label}
+          </Typography>
+        )}
```

Comment:
@noahonyejese I have two comments here:
- I think we should use the same typography styles as for other legend titles, they look consistent in the design (looks like it's a caption style, but I'd double check to make sure we use the same style in e.g. Y axis title in the code),

<img width="294" alt="Screenshot 2025-02-12 at 15 21 41" src="https://github.com/user-attachments/assets/9c8254ae-3ef4-4c4e-b86e-605bafa4c9b7" />

- we still miss the details panel wrapper that would open the panel when clicking, you can see what happens when you click the Y axis title. I'd check how it's implemented there and use the same approach 👍 

<img width="712" alt="Screenshot 2025-02-12 at 15 23 15" src="https://github.com/user-attachments/assets/81c31147-a8ce-44a6-a2fb-805ecb98d2d1" />